Implement ntcore 2018 changes

virtuald opened the pull request:
Here's the initial updates. I haven't really done anything to the user facing APIs yet... so that remains to be done. Unfortunately, this took about a week longer than I had expected. I would appreciate any initial reviews/testing of this, there's quite a few changes -- but the tests finally pass!

- Fixes #49

auscompgeek commented on tests/test_entry_notifier.py:
```diff
@@ -0,0 +1,301 @@
+''' Open Source Software - may be modified and shared by FRC teams. The code   '''
+''' must be accompanied by the FIRST BSD license file in the root directory of '''
+''' the project.                                                               '''
+'''----------------------------------------------------------------------------'''
```
Shouldn't this copyright header be comments before the future statement?

virtuald replied:
Nope, __future__ needs to be before anything else (except comments), and these aren't comments. Technically, there's no reason they have to be this way, they could be converted to comments.

virtuald commented:
Finally got around to finishing this. Added the user facing APIs, moved some things around, removed some deprecated stuff, and generally tried to match ntcore when it made sense.

One big thing is much of the old connection API stuff is gone, so if you were doing something special then it will break for you. However, I retained the 'initialize' function, which all of the examples use anyways.

Another big thing is that the server doesn't automatically get initialized, so you have to call either startServer() or initialize to start it.

.. oh, I forgot to add a test. Will fix that, then this is done.
